packages-backend/express/src/auth.ts

```diff
@@ -149,9 +155,15 @@ function createSessionAuthVerifier<
       issuer = mapToLegacyIssuer(cloudIdentifierHeader) ?? issuer;
     }
 
-    // @ts-ignore: the node HTTP request does not know about `originalUrl`
-    const requestUrlPath = request.originalUrl ?? request.url;
-    const audience = getConfiguredAudience(options, requestUrlPath);
+    const requestUrlPath = options.getRequestUrl
+      ? options.getRequestUrl(request)
+      : request.originalUrl ?? request.url;
+
+    if (!requestUrlPath || requestUrlPath === 'undefined') {
```

Comment:
Comparision with `undefined` string is to take into account `getRequestUrl` implementations like this:
```
const sessionAuthVerifier = createSessionAuthVerifier({
  audience: 'https://my-api-server.com',
  issuer: CLOUD_IDENTIFIERS.GCP_EU,
  getRequestUrl: (event) => `${event.rawPath}${event.rawQueryString ? '?' + event.rawQueryString : ''}`
});
```
It would return that `'undefined'` string and I think that implementation would be common.

Reply:
This is something I realized when writing the tests.

Let's say you `request` _like_ object has the properties `rawPath` and `rawQueryPath` but you incorrectly think the properties are called `path` and `queryPath`.
Also, let's say you implement a simple `getRequestUrl` function like this (because it seems good enough):
```
getRequestUrl: (event) => `${event.path}${event.queryString ? '?' + event.queryString : ''}`
```
Now the problem is that executing that function will yield the string `'undefined'` as its result.

Instead of raising the "Invalid request path." error, as the url is not _falsy_, this value will be passed to the `express-jwt` package and you will get an "jwt audience invalid" error.

I thought that simple implementation would be quite common and that's why I opted for extending the condition.

I know this is a controversial and maybe personal taste so I can change it if we don't feel comfortable with it.